### PR TITLE
Add instructions for running in ConductR

### PR DESCRIPTION
**Pull request by TimMoore**
Also includes fixes for a few minor ConductR compatibility issues.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -191,7 +194,16 @@ def elasticsearch : String = {
...
+  javacOptions in Compile ++= Seq("-encoding", "UTF-8", "-source", "1.8"),
+  javacOptions in (Compile, compile) ++= Seq("-Xlint:unchecked", "-Xlint:deprecation", "-parameters")
+)
+
+// Include this into impl projects that use the message broker API
+// It overrides the production configuration to use a hardcoded Kafka broker
+// host and port rather than looking it up from the service locator.
+// See docs/running-in-conductr.md for details.
+def kafkaSettings: Seq[Setting[_]] = Seq(
+  BundleKeys.startCommand += "-Dlagom.broker.kafka.service-name=''"
```
Comment (ignasi35) [inline build.sbt]: 👍

Comment (ignasi35) [inline build.sbt]: Don't we need a broker list too ?

Comment (TimMoore) [inline build.sbt]: It happens to default to localhost:9092: https://github.com/lagom/lagom/blob/master/service/core/kafka/client/src/main/resources/reference.conf#L4

Comment (ignasi35) [inline build.sbt]: How about making the broker explicit here with a note:

```
// You may have to edit this list if your Kafka Docker container is 
// not listening on 127.0.0.1:9092
```

I'd rather add this info here than in the MD file with instructions you introduced in this PR to keep that file at a minimum. Also, when the kafka bundle for ConductR is ready it'll be possible to come back here and remove this whole setup without changing other files.

WDYT?

Comment (TimMoore) [inline build.sbt]: Sure

File: search-api/src/main/java/com/example/auction/search/api/SearchService.java [hunk 2]
```diff
@@ -22,6 +22,6 @@ default Descriptor descriptor() {
                 Service.restCall(Method.GET, "/search?pageNo&pageSize", this::search)
         ).withPathParamSerializer(
                 UUID.class, PathParamSerializers.optional("UUID", UUID::fromString, UUID::toString)
-        ).withAutoAcl(true);
+        );
```
Comment (ignasi35) [inline search-api/src/main/java/com/example/auction/search/api/SearchService.java]: 👏

File: build.sbt [hunk 3]
```diff
@@ -203,7 +203,12 @@ def commonSettings: Seq[Setting[_]] = eclipseSettings ++ Seq(
 // host and port rather than looking it up from the service locator.
 // See docs/running-in-conductr.md for details.
 def kafkaSettings: Seq[Setting[_]] = Seq(
-  BundleKeys.startCommand += "-Dlagom.broker.kafka.service-name=''"
+  BundleKeys.startCommand ++= Seq(
+    "-Dlagom.broker.kafka.service-name=''",
+    // You may have to edit this list if your Kafka
+    // server is not listening on 127.0.0.1:9092
+    "-Dlagom.broker.kafka.brokers='127.0.0.1:9092'"
+  )
```
Comment (ignasi35) [inline build.sbt]: After merging I realised that I?m not sure this is the recommended approach for production users to override Kafka settings. 
Should we document this as a best practice in the docs for the use case where a developer wants to override the settings, or should we make it obvious here that this is a hack to ease the usage of the `online-auction` sample app?
see https://github.com/lagom/lagom/issues/702

Comment (TimMoore) [inline build.sbt]: Do you mean you're not sure if setting it in this particular ConductR setting is the recommended approach?

I'm not sure, either. When I asked about it in Slack, I didn't get any responses, and the Play guide offered a number of different options but didn't talk about ConductR specifically.

I based this approach on a little snippet of code I found in the [`sbt-conductr` README](https://github.com/typesafehub/sbt-conductr#adding-start-command-options). I'm not sure if this is best practice.

If it were more than a small number of properties, I would probably want to put them into a separate `.conf` file and pass `-Dconfig.resource` instead, as described in https://www.playframework.com/documentation/2.5.x/ProductionConfiguration#Specifying-an-alternate-configuration-file

I think #702 will be a better way to handle this in the long run.

**Pull request merged**